### DISCUSSION TITLE
Best way to run a script file and exit the CLI on Windows?

### DISCUSSION THREAD

**Discussion by DpadamsX** [duckdb/duckdb#14679] [Q&A]
I'm using the DuckDB.exe CLI tool, and want it to read and process a script file that I've prepared and then exit. I'm running a line like this:

```
C:\Ascendco\Scripts\DuckDB\duckdb.exe -init  C:\Users\estes\AppData\Local\Temp\DuckDB_03070A43A136D54E9B0D703971D678ED.sql
```

I kind of feel like I'm abusing `-init` here, is there _another_ way to invoke the CLI to run a script file and push the output to `STDOUT`?

As an example, here's my "hello world" test script in `TempDuckDB_0E68CAD4296AFD44920B75C7001A9056.sql`:

```
.mode csv
.headers no

SELECT 'File says that the current time is ' || now();
```

DuckDB 1.1.2 is installed, paths are fine, code is fine, works on macOS with posix paths fine.

If I use the Task Manager to kill off the DuckDB shell that's been launched (but not shown), the waiting program returns **with the right results**.

I'm not invoking DuckDB directly from the command line, I'm doing so from within an (obscure) tool named 4D. It's capable of pushing calls to the command line synchronously. However, in this case, it's behaving like it's launched a GUI program, and is waiting for it to exist. No, it doesn't seem like a DuckDB problem, but I'm hoping for a DuckDB solution.

I've tried adding `.quit` to the script file, no difference. I've tried `.exit` without a code, as I haven't tracked down what the codes should be.

My best hope is that there is a way to set the command line flags to `Run file and exit. Hard.`

Grateful for any help, thanks.

**Comment by psanker** [comment]
Use `duckdb.exe < path\to\script.sql` in cmd

**Comment by DpadamsX** [comment]
Thank you! I did get it sorted out, and should have posted sooner, I'll leave a few more notes for anyone else.

Yes, you need to get the path right, I like the idea fro psanker of directing the script path at the binary. 

In my case, I ended up escaping the full path on Windows, which works. I'm using a full path as that makes no assumptions about how the environment is configured, and makes it simpler to spot references to the wrong copy of `ducked.exe`. At least it does for me.

On Windows, I found an example in the `tldr` output that works well. The `-c COMMAND` from the docs didn't clarify the `-c` flag for me, I didn't know what belongs there. Turns out, it  is `-C "ducked commands"`, like:

```
- Run an SQL script:
    duckdb -c ".read path/to/script.sql"
```

For example, on macOS this time:

```
/Users/bob/duckdb -json -noheader -c ".read /var/folders/gk/n4942dl177j4_9334nm78p340000gq/T/DuckDB_IB_Scratch/DuckDB_F49BC585BBAA486DAC233932FB3C6838.sql"
```
